### PR TITLE
Add search action to pages

### PR DESCRIPTION
**Pull request by zanderle**
Once the API will support search for pages, this will enable it through client as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,9 @@ client.pages.get('news', 'hello-world')
 client.pages.get('news', 'hello-world', {'foo': 'bar'})
 ```
 
+```python
+client.pages.search('enter search query', {'page': 1, 'page_size': 10})
```
Comment (orlyohreally) [inline README.md:77]: > The Page's `.all()` and `.get()` methods accept an optional `params` to add additional data to the response.

 Should probably update this line to include `search` method

File: butter_cms/page.py [hunk 2]
```diff
@@ -14,3 +14,13 @@ def all(self, page_type, params=None):
     def get(self, page_type, page_slug, params=None):
         full_slug = '{}/{}/'.format(page_type, page_slug)
         return self.api_get(slug=full_slug, params=params)
+
+    def search(self, query, params=None):
```
Comment (orlyohreally) [inline butter_cms/page.py:18]: Why not set default value for `params` to be `{}`?

**Review by orlyohreally — COMMENTED** [review]
Works as expected, just small notes

File: butter_cms/page.py (hunk 2 above)
Comment (zanderle) [reply butter_cms/page.py:18]: The default values for function arguments should never be mutable (like `[]` or `{}`), as you can get unexpected behaviour that way.

**Pull request merged**